src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -207,6 +251,56 @@ int IVsEditorFactory.Close()
             return LoaderName;
         }
 
+        private (string, Guid, IntPtr) GetWinFormsDesignerData(IVsHierarchy vsHierarchy, uint itemid, IVsTextBuffer? textBuffer, READONLYSTATUS readOnlyStatus, bool initializeView)
+        {
+            var loaderName = GetWinFormsLoaderName(vsHierarchy);
+            var designerService = (IVSMDDesignerService)_oleServiceProvider.QueryService<SVSMDDesignerService>();
+            var designerLoader = (IVSMDDesignerLoader)designerService.CreateDesignerLoader(loaderName);
+            if (designerLoader is null)
+            {
+                return GetCodeEditorData(textBuffer, readOnlyStatus, initializeView);
+            }
+
+            try
+            {
+                if (initializeView)
+                {
+                    designerLoader.Initialize(_oleServiceProvider, vsHierarchy, (int)itemid, textBuffer as IVsTextLines);
+                }
+                var editorCaption = designerLoader.GetEditorCaption((int)readOnlyStatus);
+
+                var designer = designerService.CreateDesigner(_oleServiceProvider, designerLoader);
+
+                var documentView = initializeView ? Marshal.GetIUnknownForObject(designer.View) : IntPtr.Zero;
+                var commandUI = designer.CommandGuid;
+
+                return (editorCaption, commandUI, documentView);
+            }
+            catch
+            {
+                designerLoader.Dispose();
```

Comment:
You only want to dispose `designerLoader` if there's an exception?  Not always?  That's strange enough to warrant a comment.

Reply:
This code is pulled out from IVsEditorFactory.CreateEditorInstance() (line 152 in the base version). I don't want to change the working old code. Maybe in a subsequent change by the Roslyn change.

Reply:
I tried to dispose of it in a finally block. We get errors at a later point. Any fix to this code is best handled by the Winforms team.

  <entry>
    <record>4002</record>
    <time>2020/11/06 23:27:14.539</time>
    <type>Error</type>
    <source>Editor or Editor Extension</source>
    <description>System.ObjectDisposedException: Cannot access a disposed object.&#x000D;&#x000A;Object name: &apos;VSCodeDomDesignerLoader&apos;.&#x000D;&#x000A;   at System.ComponentModel.Design.Serialization.BasicDesignerLoader.get_LoaderHost()&#x000D;&#x000A;   at System.ComponentModel.Design.Serialization.BasicDesignerLoader.OnEndLoad(Boolean successful, ICollection errors)&#x000D;&#x000A;   at Microsoft.VisualStudio.WinForms.RemoteClient.Loader.RemoteCodeDomDesignerLoader.OnEndLoad(Boolean successful, ICollection errors)&#x000D;&#x000A;   at System.ComponentModel.Design.Serialization.BasicDesignerLoader.System.ComponentModel.Design.Serialization.IDesignerLoaderService.DependentLoadComplete(Boolean successful, ICollection errorCollection)&#x000D;&#x000A;   at Microsoft.VisualStudio.Design.Serialization.CodeDom.VSCodeDomDesignerLoader.DeferredLoadHandler.Microsoft.VisualStudio.TextManager.Interop.IVsTextBufferDataEvents.OnLoadCompleted(Int32 fReload)&#x000D;&#x000A;   at Microsoft.VisualStudio.Editor.Implementation.TextDocData.&lt;&gt;c__DisplayClass180_1.&lt;RaiseOnLoadCompletedEvent&gt;b__0()&#x000D;&#x000A;   at Microsoft.VisualStudio.Text.Utilities.GuardedOperations.CallExtensionPoint(Object errorSource, Action call)&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)&#x000D;&#x000A;--- End of stack trace from previous location where exception was thrown ---&#x000D;&#x000A;   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)</description>
  </entry>

Reply:
Don't dispose the designer loader! The `IVSMDDesignerService.CreateDesigner(...)` method in VS passes it into the `DesignSurface` that gets created and is used to perform the actual load (and reloads -- which happen during normal designer operation). http://index/?leftProject=Microsoft.VisualStudio.Design&leftSymbol=n8p1tszkfyz7&file=DesignerActivationService.cs&line=629

Reply:
def doc this conversation in the code.

Reply:
Absolutely, let's doc this.